birdhouse/optional-components/secure-thredds/secure-access-magpie-permission.cfg

```diff
@@ -1,3 +1,9 @@
+users:
+  - username: authtest
+    password: authtest1234
+    email: authtest@example.com
+    group: anonymous
```

Comment:
We already create this same test user here https://github.com/bird-house/birdhouse-deploy/blob/edd8ab40f783b6b724b88c1abd5c2ac1d721f8dc/birdhouse/scripts/create-magpie-authtest-user#L13-L19, so why repeat it here again?

Also the component is called `secure-thredds`, so in this spirit, adding a publicly known user and password is not so "secure".

Granted the same publicly known user and password is in the `create-magpie-authtest-user` but at least the script says test user in there and do not imply anything "secure".

Reply:
I added this by request of @MatProv to fix tests that seemed to have mismatches between the created user. 
He should be able to detail more about the problems he faced.

I do agree that adding it in the file doesn't make it "secure", but the user was already referenced in other permissions before, so it wasn't secure in the first place anyway. 

The best approach would be instead to create some kind of "thredds-secure" group, to which "authtest" user is applied membership when created via [create-magpie-authtest-user](https://github.com/bird-house/birdhouse-deploy/blob/edd8ab40f783b6b724b88c1abd5c2ac1d721f8dc/birdhouse/scripts/create-magpie-authtest-user#L13-L19) instead of "anonymous". That "thredds-secure" group would replace all the `user: authtest` in that config, so "authtest" user would inherit the group permissions. Not adding the creation of "authtest" user would still make the "thredds-secure" exist, but no user would be part of it. That would also facilitate future users we want to grant access to `secure` directory, by simply adding them to that group instead of duplicating all the permissions for each one.

Reply:
@tlvu  So the issue we had was that the permissions were attributed to the authtest user before it has been created. The result was that magpie created a default user named authtest with a dummy password, added the permissions and trying to use authtest in the notebook failed since the username/password we specify later on in `create-magpie-authtest-user` wasn't used. Now, specifying the password directly in the cfg makes sure the password isn't a dummy one. Now the failed 401 test (`pavics_thredds.ipynb`) passes.

The `subset.ipynb` is still failing but i'm pretty sure it's unrelated to magpie.

Reply:
>  permissions were attributed to the authtest user before it has been created.

@MatProv Ah OK I understand now.  

@fmigneault I agree if we have the `authtest-group` instead, then the user creation won't happen so then the `create-magpie-authtest-user` script will be able to create the user with the expected password and we won't have to hardcode the user creation here.

The `create-magpie-authtest-user` will have to create that `authtest` user part of group `authtest-group` instead of group `anonymous` like right now.

Can you guys do this?  I am guessing just replace `user: authtest` by `group: authtest-group` in this file `secure-access-magpie-permission.cfg` and update the group in `create-magpie-authtest-user`?

Reply:
@tlvu  
Sure! I will push the update with the group. 
I will call the group `thredds-secure` because there is no reason to make it specific to `authtest` (the `secure-access-magpie-permission.cfg` shouldn't have any knowledge of `authtest` on its own).

Reply:
All done.

Reply:
> I will call the group `thredds-secure` because there is no reason to make it specific to `authtest` (the `secure-access-magpie-permission.cfg` shouldn't have any knowledge of `authtest` on its own).

@fmigneault oops missed this reply.  With the explicit naming, I just want to prevent users to re-use this group for other purposes than testing, else the `authtest` user will have access to more than it should and that would be a security leak.